src/Internal/ConverterProvider.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace Retrofit\Internal;
+
+use Retrofit\Converter\Converter;
+use Retrofit\Converter\ConverterFactory;
+use Retrofit\Type;
+use RuntimeException;
+
+readonly class ConverterProvider
+{
+    /** @param ConverterFactory[] $converterFactories */
+    public function __construct(private array $converterFactories)
+    {
+    }
+
+    public function getRequestBodyConverter(Type $type): Converter
+    {
+        foreach ($this->converterFactories as $converterFactory) {
```

Comment:
Duplicated code (three times). It should be one method with two params -> Closure and exception message.

Reply:
By the way, I can see we use Ouzo, so `Arrays::any` should be handy here :)

Reply:
You meant `Arrays::find` ofcourse?